projectdoc/implementation.tex

```diff
@@ -270,8 +270,45 @@ \subsubsection{Umsetzung Spezifikation}
 \paragraph{Berechnung der Haltestellenkategorie}~\\
 TODO
 
-\paragraph{Berechnung der Gehezeit zur Haltestelle}~\\
-TODO
+\paragraph{Berechnung der Gehzeit zur Haltestelle}~\\
+
+\subparagraph{Snapping von der Haltestelle auf den Routing-Graph}
+Die Haltestellen stammen aus dem Datenstamm der SBB~\cite{sbb_hafas_spec} und sind somit nicht Teil des Routing-Graphen. 
+Um die \glspl{Isochrone} berechnen zu können, muss auf dem Routing-Graphen zuerst ein Startpunkt ermittelt werden, von dem das Routing entlang der Strassen und Wege ausgehen kann.
+Dieser Punkt soll möglichst nahe an der eigentlichen Haltestelle liegen.
+Es ist anzumerken, dass unsere Fahrplan-Daten lediglich ein geographischer Punkt pro Haltestelle definiert und so Buskanten oder Gleise nicht genau abbilden.
+
+Ein erster Ansatz dieses "`Snappings"' sucht den geographisch nächstliegenden Vertex des Routing-Graphen zu den Koordinaten der Haltestelle.
+Dies ist bei regulären Bushaltestellen mit zwei Kanten ein guter Ansatz, da diese Koordinaten meist neben oder direkt auf der Strasse liegen.
+Problematisch wird dies aber etwa bei grossen Bahnhöfen wie dem Zürcher Hauptbahnhof.
+Die Koordinate des Bahnhofs liegt in der Mitte der Bahnhofshalle, welche als Fussgängerfläche erfasst ist.
+Da solche Flächen nicht im Routing-Graphen abgebildet werden, liegt der nächste Vertex des Routing-Graphen auf einer Rolltreppe, die nicht mit dem restlichen Graphen verbunden ist (siehe Abbildung~\ref{fig:snapping_comparison}).
+Dieses Snapping auf einen isolierten Graphen führt dazu, dass die Routing-Engine nie ausserhalb des Bahnhofs routen kann, dadurch können auch keine \glspl{Isochrone} erzeugt werden.
+
+\begin{figure}[ht]
+    \centering
+    \includegraphics[width=1\linewidth]{projectdoc/img/snapping_comparison}
+    \caption[Snapping der Haltestelle auf den Routing-Graphen]{Die Haltestelle (blaues Quadrat) wird mit dem einfachen Ansatz zur Rolltreppe `"gesnapped"' (Roter Diamant), während die Optimierung (Grüner Punkt) ein Ergebnis erzeugt, dass mit dem restlichen Routing-Netz verbunden ist.}
+    \label{fig:snapping_comparison}
+\end{figure}
+
+Ein optimierter Ansatz nimmt nicht direkt den nächstgelegen Punkt auf dem Routing-Graphen, sondern prüft, ob der ermittelte Vertex auf einem isolierten Segment des Graphen liegt.
+Dafür werden mit Hilfe der Routing-Engine alle Punkte gesucht, die auf dem Graphen in 200 Metern Fussweg erreichbar sind.
```

Comment:
Hier ist noch nicht klar ersichtlich, dass innerhalb von 200 Meter eine nearest neighbour search durchgeführt und von welchem Punkt aus dann die drivingDistance generiert wird.
Mir kam leider auch keine bessere Formulierung in den Sinn.

Reply:
Habs noch etwas anders formuliert. Es ist aber nicht richtig, dass innerhalb von 200 Metern eine nearest-neighbor-Suche gemacht wird. Die 200m gelten für die Drivingdistance. Die Iteration geht einfach die nearest neighbors der Reihe nach durch (limitiert auf 100)